Rubberduck.RibbonDispatcher/AbstractRibbon.cs

```diff
@@ -0,0 +1,67 @@
+        }
+        protected RibbonFactory     RibbonFactory    { get; private set; }    // A property because might be exposed in future
+
+        public IRibbonCommon     Controls(string controlId)              => GetValueOrNull(RibbonFactory.Controls, controlId);
+        public IRibbonButton     Buttons(string controlId)               => GetValueOrNull(RibbonFactory.Buttons,  controlId);
+        public IRibbonToggle     Toggles(string controlId)               => GetValueOrNull(RibbonFactory.Toggles,  controlId);
+        public IRibbonDropDown   DropDowns(string controlId)             => GetValueOrNull(RibbonFactory.DropDowns,controlId);
+
+        // All controls (almost) including Groups
+        public string            GetDescription(IRibbonControl control)  => Controls(control?.Id)?.Description??Unknown(control);
+        public string            GetKeyTip(IRibbonControl control)       => Controls(control?.Id)?.KeyTip     ??"??";
+        public string            GetLabel(IRibbonControl control)        => Controls(control?.Id)?.Label      ??Unknown(control);
+        public string            GetScreenTip(IRibbonControl control)    => Controls(control?.Id)?.ScreenTip  ??Unknown(control);
+        public string            GetSuperTip(IRibbonControl control)     => Controls(control?.Id)?.SuperTip   ??Unknown(control);
+
+        public bool              GetEnabled(IRibbonControl control)      => Controls(control?.Id)?.Enabled    ??false;
+        public IPictureDisp      GetImage(IRibbonControl control)        => Controls(control?.Id)?.Image;
+        public RibbonControlSize GetSize(IRibbonControl control)         => Controls(control?.Id)?.Size       ??RibbonControlSizeLarge;
+        public bool              GetVisible(IRibbonControl control)      => Controls(control?.Id)?.Visible    ??true;
+
+        // Button Controls
+        public bool              GetShowImage(IRibbonControl control)    => Buttons(control?.Id)?.ShowImage   ??false;
+        public bool              GetShowLabel(IRibbonControl control)    => Buttons(control?.Id)?.ShowLabel   ??true;
+        public void              OnAction(IRibbonControl control)        => Buttons(control?.Id)?.OnAction();
+
+        // Toggle Controls: checkBoxes & toggleButtons
+        public bool GetPressed(IRibbonControl control)                   => Toggles(control?.Id)?.IsPressed   ??false;
+        public void OnActionToggle(IRibbonControl control, bool pressed) => Toggles(control?.Id)?.OnAction(pressed);
+
+        private static TValue GetValueOrNull<TValue>(IReadOnlyDictionary<string,TValue> dictionary, string key)
+        {
+            TValue ctrl;
+            dictionary.TryGetValue(key, out ctrl);
+            return ctrl;
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Globalization", "CA1305:SpecifyIFormatProvider", MessageId = "System.String.Format(System.String,System.Object)")]
```

Comment:
the codebase does not use these MessageSuppression attributes anywhere else IIRC

Reply:
I am not sure I get your point. I have always turned "All Microsoft Rules" on, as I find that about 3% to 5% of generated warnings identify potential bugs. But that means all the noise must be suppressed in order to see the signal.  Do you have a better way?

Reply:
The point I'm trying to make (just like above with the uniformity and sameness) is that the code you PR should be similarly styled and similarly architected as the rest of the codebase.

I found that a uniform styling and uniform use of features is making it easier to skim large codebases. This includes things like [SuppressMessage], but also more "trivial" matters like basic formatting rules.
FWIW I like the idea of having all Diagnostic rules turned on, but I prefer to not add even more clutter to the code by suppressing those warnings.